proto/public/uber/cadence/api/v1/common.proto

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package uber.cadence.api.v1;
+
+option go_package = "apiv1";
+option java_multiple_files = true;
+option java_outer_classname = "ApiProto";
+option java_package = "com.uber.cadence.api.v1";
+
+import "google/protobuf/duration.proto";
+import "dependencies/gogoproto.proto";
+
+message WorkflowExecution {
+  string workflow_id = 1;
+  string run_id = 2;
+}
+
+message WorkflowType {
+  string name = 1;
+}
+
+message ActivityType {
+  string name = 1;
+}
+
+message Payload {
+  bytes data = 1;
+}
+
+message Failure {
+  string reason = 1;
+  bytes details = 2;
+}
+
+message Memo {
+  map<string, Payload> fields = 1;
+}
+
+message Header {
+  map<string, Payload> fields = 1;
+}
+
+message SearchAttributes {
+  map<string, Payload> indexed_fields = 1;
+}
+
+enum EncodingType {
+  ENCODING_TYPE_INVALID = 0;
+  ENCODING_TYPE_THRIFTRW = 1;
+  ENCODING_TYPE_JSON = 2;
+  ENCODING_TYPE_PROTO3 = 3;
+}
+
+message DataBlob {
+  EncodingType encoding_type = 1;
+  bytes data = 2;
+}
+
+message WorkerVersionInfo {
+  string impl = 1;
+  string feature_version = 2;
+}
+
+message SupportedClientVersions {
+  string go_sdk = 1;
+  string java_sdk = 2;
+}
```

Comment:
probably worth documenting the format / how comparison is defined.  I'm _guessing_ it's a standard semver string... but in that case string comparison won't get the right result, as `1.2.3` is less than `1.2.3-rc1`, so hopefully it's something else.

Reply:
Yes, this is standard semver. We use https://github.com/hashicorp/go-version for comparison.